### PR TITLE
ref(ui): Refactor PanelBody to have simpler props

### PR DESCRIPTION
**Pull request by BYK**
This patch renames the padding-related prop to avoid the double negative and removes the never-used `direction` prop.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/panels/panelBody.tsx [hunk 1]
```diff
@@ -9,33 +9,34 @@ import textStyles from 'app/styles/text';
 type FlexComponentProps = Omit<React.ComponentPropsWithoutRef<typeof Flex>, 'theme'>;
 
 type Props = FlexComponentProps & {
-  disablePadding?: boolean;
-  direction?: FlexComponentProps['flexDirection'];
+  flexible?: string;
```
Comment (BYK) [inline src/sentry/static/sentry/app/components/panels/panelBody.tsx]: Converting these to strings was a mere TypeScript hiccup and I'll revert this.

File: src/sentry/static/sentry/app/components/panels/panelBody.tsx [hunk 2]
```diff
@@ -9,33 +9,34 @@ import textStyles from 'app/styles/text';
 type FlexComponentProps = Omit<React.ComponentPropsWithoutRef<typeof Flex>, 'theme'>;
 
 type Props = FlexComponentProps & {
-  disablePadding?: boolean;
-  direction?: FlexComponentProps['flexDirection'];
+  flexible?: boolean;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/panels/panelBody.tsx]: What about `isFlex`

File: src/sentry/static/sentry/app/components/panels/panelBody.tsx [hunk 3]
```diff
@@ -9,33 +9,34 @@ import textStyles from 'app/styles/text';
...
+  withPadding?: boolean;
+  flexDir?: FlexComponentProps['flexDirection'];
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/panels/panelBody.tsx]: What's wrong with `direction`? Do we even use it?

**Review by billyvg — COMMENTED** [review]
Did you check if `getsentry` uses this component?

File: src/sentry/static/sentry/app/components/panels/panelBody.tsx (hunk 2 above)
Comment (BYK) [reply src/sentry/static/sentry/app/components/panels/panelBody.tsx]: `<PanelBody flexible withPadding>` vs `<PanelBody isFlexible withPadding>`. The former felt more "natural" to me but I'm not too vested in the name. We just cannot use `flex` as it clashes with existing props on `FlexComponentProps`.

File: src/sentry/static/sentry/app/components/panels/panelBody.tsx (hunk 3 above)
Comment (BYK) [reply src/sentry/static/sentry/app/components/panels/panelBody.tsx]: For some reason TypeScript was also complaining that it colliding with something on `FlexComponentProps` or props in `FlexBox`. It doesn't anymore so maybe that was a false alarm? It doesn't seem to be used anywhere though so maybe I should just remove it?

**Review by BYK — COMMENTED** [review]
>Did you check if getsentry uses this component?

Nope, good point. Will check and follow up here.

**Comment by billyvg** [comment]
Tagging @EvanPurkhiser since I think he's looked into this a bit too

**Comment by EvanPurkhiser** [comment]
Love the change on disablePadding, I’m confused what the issue was with direction? I wasn’t having any typescript errors? Or did this cause more types to be generated due to discriminate typing?

File: src/sentry/static/sentry/app/components/panels/panelBody.tsx (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/panels/panelBody.tsx]: Yes to removing it

**Comment by BYK** [comment]
>I’m confused what the issue was with direction? I wasn’t having any typescript errors? Or did this cause more types to be generated due to discriminate typing?

I got this while I was working on backporting the emotion types. I think it was a bug during development so that point is moot now.

**Review by billyvg — APPROVED** [review]
Percy looks good, I did a grep for `styled(PanelBody)`, but doesn't look like `direction` was in use.

**Pull request merged**